### PR TITLE
Proposed code of conduct changes

### PR DESCRIPTION
**Pull request by Quinncuatro**
Happy December, everyone!

Our mission with NewHaven.io is to grow the tech community in the greater New Haven area. We on the board want to provide a safe, inclusive, and open environment in which all people are welcome to share their ideas, thoughts, and experiences so that we can all learn from each other and be better for it.

Those of you who have been here for a time know that we’ve had very few problems with the conduct of IO’s members over the years, but as our numbers keep growing and as we’re getting ready to move into a new year we (I, and the rest of the board) thought it would be prudent to make sure our CoC amounts to more than a boilerplate “don’t be a jerk” spiel.

Among the changes, you’ll find new sections on expected and unacceptable behavior, consequences of said unacceptable behavior and reporting guidelines, and a grievance process for members to use when they feel they’ve been given an unfair ruling on a grievance directed at them.

We’ve borrowed heavily from the folks at [Denver Devs](https://denverdevs.com), [Geek Feminism](http://geekfeminism.wikia.com), and the [Citizen Code of Conduct](http://citizencodeofconduct.org/) because we believe they (and the various groups they’re commingled with) are setting excellent examples of how tech communities (and communities in general) should be run (wholly inclusive, supportive, and professional), especially when taking into account how much privilege we all hold as tinkerers, workers, and managers in tech.

The board feels strongly about our guiding principle of “be excellent to each other” and hope that the community that all of us has helped foster resonates with what we’re putting in this PR.

That said, in the spirit of open source citizenship (read the PR!), we wanted to open up a discussion for the month of December to make sure we got this right. We want to continue fostering a community where everyone from green hobbyists to the most experienced specialists feel welcome to join in on the discussion, ask questions, and learn from each other.

IO wouldn’t be what it is without all of you contributing to Slack, our Meetup events, and the general tone of the tech scene in New Haven. We hope that you’ll bring that same candor to these proposed changes to our Code of Conduct and help us catch gaps we may have missed.

This will remain open until the end of December, with changes happening along the way as things are discussed, so that we can go into the new decade with a fresh set of rules to help us continue to grow as an organization.

Be excellent to each other.

### CODE REVIEW

**Review by nrawling — APPROVED** [review]
I like it. Thank you all for working so hard to make this a safe (and productive, useful, thoughtful, engaging) space for all of us.

**Comment by gwright** [comment]
I question the need to call out particular characteristics when describing inappropriate behavior.  For example:

> In that vein, we want to provide a harassment-free experience for everyone, regardless of gender identity or expression, sexual orientation, disability, physical appearance, body size, race, age, religion, or non-religion.

Couldn't this just be:
>  In that vein, we want to provide a harassment-free experience for everyone.

The danger of listing things in a document like this is that you'll always leave something out.  What about nationality? ethnicity? political affiliation?  tabs-vs-spaces?  There are several places where lists like this appear in the doc and they are all slightly different. If there is a desire to keep the enumerations, perhaps dry up the document so the list is only needed once or make sure the list is identical in all appearances.

**Comment by gwright** [comment]
There are a couple references to actions outside the scope of newhaven.io activities. For example:
> This code of conduct and its related procedures also applies to unacceptable behavior occurring outside the scope of community activities when such behavior has the potential to adversely affect the safety and well-being of community members.

I am ambivalent about this scope. It isn't clear to me that newhaven.io should extend its concern to include interactions between people outside of its forums/activities. How is newhaven.io supposed to evaluate actions taken outside its purview or accusations unrelated to newhaven.io? Do we want to participate in "cancel culture"?

**Comment by josh-levinson** [comment]
@gwright I think there is a case to be made that certain behaviors are egregious enough to warrant expulsion from the community. For instance, if a member committed a felony assault on someone, even if it wasn't at an event we sponsored, we'd still consider that person a possible risk to our members.

Perhaps that process needs to be more formalized, i.e., the Board Members consider it and, y'know, let's say someone has not done anything bad for a long time, consider reconstitution, but I do think there are certainly actions members could take outside of newhaven.io which would warrant expulsion for the safety of everyone.

**Comment by retrojayne** [comment]
@gwright I think the gist of this is that we are trying to be very explicit about being welcoming and supportive as well as having processes in place to handle situations of harassment or otherwise uncivil or unfriendly behavior. When people consider joining New Haven IO, they will know that we value open source citizenship and we support all of our members. Hopefully, that will encourage even more people to join.

**Comment by maxx1128** [comment]
I agree with @jlevnhv and @retrojayne here, they make good points about the scope and explicitness of behaviors that won't be accepted. In my opinion, it's better this way since it leaves less room for inappropriate or unwelcoming behavior to wiggle in.

**Comment by gwright** [comment]
I can't argue with the idea that the group should be welcoming and harassment free. I just don't think particular flavors of harassment need to be highlighted.  There should probably be some sort of "including but not limited to ..." language though if that approach is taken.

**Comment by Quinncuatro** [comment]
Agreed on the "but not limited to" bit, @gwright.

**Comment by gwright** [comment]
@jlevnhv I don't think the problem comes from a scenario as blatant and illegal as you describe but instead comes from the more ambiguous scenarios where one person is using the system to hurt someone else without there being any legal/criminal infraction.  How is newhaven.io going to adjudicate claims about something that happened external to the organization?  Why should it? Who's side should newhaven.io take?  If there is an actual safety issue, shouldn't the police/courts being dealing with it (restraining orders, arrests, etc.)?

**Comment by lourinaldi** [comment]
y'all cray.

File: README.md [hunk 1]
```diff
@@ -1,16 +1,98 @@
...
+## Contact info
+
+Feel free to contact any of the NewHavenIO board members at any time. You'll likely get a faster response on Slack than by e-mail.
+
+Henry Quinn (@henry)
+Jenn Nolte (@Jenn Nolte she/her)
+Josh Levinson (@jlevnhv)
+Steve Voils (@steve)
+Tom Reznick (@tom)
+Ben Mindyette (@ben)
```
Comment (bmidi) [inline README.md]: Typo here (Midyette)

**Review by bmidi — CHANGES_REQUESTED** [review]

**Comment by retrojayne** [comment]
@gwright because if there is someone in our membership that is doing things in OR outside our community that makes members in our community feel unsafe, unwelcome, or uncomfortable, we want to know so we can deal with it early and circumvent people leaving the community because of a misunderstanding or easily resolved situation. 

It is highly unlikely that someone will go to the trouble of using our code of conduct to 'hurt' someone else in our membership, and even more unlikely that the board would be unanimously collectively deceived to that end. 

Rather, we want to make the process of dealing with the unacceptable behaviors of community members very clear so people will feel comfortable joining and comfortable speaking up - and this way, hopefully issues are resolved quickly, transparently, and justly, if issues arise at all.

**Review by Quinncuatro — COMMENTED** [review]
Good catch!

File: README.md (hunk 1 above)
Comment (Quinncuatro) [reply README.md]: Good catch!

Comment (bmidi) [reply README.md]: Although it has a ring to it... bmindy? lol

**Pull request merged**